modules/javafx.base/src/test/java/test/javafx/collections/ModifiableObservableListBaseTest.java

```diff
@@ -0,0 +1,249 @@
+        };
+
+        assertDoesNotThrow(() -> list.removeAll(Collections.<String>emptyList()));
+    }
+
+    @Test
+    public void testRetainAllWithEmptyCollectionArgumentDoesNotCallContains() {
+        var list = new MockModifiableObservableList(new ArrayList<>(List.of("a", "b", "c")));
+
+        assertDoesNotThrow(() -> list.retainAll(new ArrayList<String>() {
+            @Override
+            public boolean contains(Object o) {
+                throw new AssertionError("contains() was not elided");
+            }
+        }));
+    }
+
+    @Test
+    public void testRetainAllWithEmptyCollectionArgumentCallsClear() {
+        boolean[] flag = new boolean[1];
+        new MockModifiableObservableList(new ArrayList<>(List.of("a", "b", "c"))) {
+            @Override
+            public void clear() {
+                flag[0] = true;
+                super.clear();
+            }
+        }.retainAll(Collections.<String>emptyList());
+
+        assertTrue(flag[0], "clear() was not called");
+    }
+
+    @Test
+    public void testRetainAllOnEmptyListDoesNotCallClear() {
+        var list = new MockModifiableObservableList(new ArrayList<>()) {
+            @Override
+            public void clear() {
+                throw new AssertionError("clear() was not elided");
+            }
+        };
+
+        assertDoesNotThrow(() -> list.retainAll(Collections.<String>emptyList()));
+    }
+
+    @Test
+    public void testRemoveEmptyRangeDoesNotEnumerateList() {
+        var list = new MockModifiableObservableList(List.of("a", "b", "c")) {
+            @Override
+            public Iterator<String> iterator() {
+                throw new AssertionError("iterator() was not elided");
+            }
+
+            @Override
+            public ListIterator<String> listIterator() {
+                throw new AssertionError("listIterator() was not elided");
+            }
+        };
+
+        assertDoesNotThrow(() -> list.removeRange(0, 0));
+    }
+
+    @Nested
+    class SubListTest {
+        @Test
+        public void testAddAllWithEmptyCollectionArgumentDoesNotEnumerateCollection() {
+            var backingSubList = new ArrayList<>(Collections.<String>emptyList()) {
+                @Override
+                public Iterator<String> iterator() {
+                    throw new AssertionError("iterator() was not elided");
+                }
+
+                @Override
+                public ListIterator<String> listIterator() {
+                    throw new AssertionError("listIterator() was not elided");
+                }
+            };
+
+            var subList = new MockModifiableObservableList(Collections.emptyList(), backingSubList).subList(0, 0);
+            assertDoesNotThrow(() -> subList.addAll(Collections.emptyList()));
+            assertDoesNotThrow(() -> subList.addAll(0, Collections.emptyList()));
+        }
+
+        @Test
+        public void testAddAllWithEmptyCollectionArgumentAndInvalidIndexThrowsIOOBE() {
```

Comment:
Could use variant here with non-empty collection, and see if it still throws IOOBE.